DIPs/DIP1xxx-rval_to_ref.md

```diff
@@ -0,0 +1,345 @@
+
+The choice to receive an argument by value or by reference is a detail that the API *author* selects with respect to criteria relevant to their project or domain, however, the semantic impact is not worn by the API author, but rather by the API user, who may be required to jump through hurdles to interact the API with their local code.  
+It would be ideal if the decision to receive arguments by value or by reference were a detail for the API, and not increase the complexity of the users code.
+
+Here is proposed a strategy to emit implicit temporaries to conveniently interact with APIs that use `ref` arguments.
+
+## Reference
+
+There is a lot of prior discussion on this topic. Much is out of date now due to recent language evolution.  
+Prior discussion involving `scope`, and `auto ref` as solutions are out of date; We have implemented `scope`, `auto ref`, and we also have `return ref` now, which affects the conversation.
+
+Forum threads:  
+https://forum.dlang.org/thread/mailman.3720.1453131378.22025.digitalmars-d@puremagic.com  
+https://forum.dlang.org/thread/rehsmhmeexpusjwkfnoy@forum.dlang.org  
+https://forum.dlang.org/thread/mailman.577.1410180586.5783.digitalmars-d@puremagic.com  
+https://forum.dlang.org/thread/km4rtm$239e$1@digitalmars.com  
+https://forum.dlang.org/thread/kl4v8r$tkc$1@digitalmars.com  
+https://forum.dlang.org/thread/ylebrhjnrrcajnvtthtt@forum.dlang.org  
+https://forum.dlang.org/thread/ntsyfhesnywfxvzbemwc@forum.dlang.org  
+https://forum.dlang.org/thread/uswucstsooghescofycp@forum.dlang.org  
+https://forum.dlang.org/thread/zteryxwxyngvyqvukqkm@forum.dlang.org  
+https://forum.dlang.org/thread/yhnbcocwxnbutylfeoxi@forum.dlang.org  
+https://forum.dlang.org/thread/tkdloxqhtptpifkhvxjh@forum.dlang.org  
+https://forum.dlang.org/thread/mailman.1478.1521842510.3374.digitalmars-d@puremagic.com  
+https://forum.dlang.org/thread/gsdkqnbljuwssslxuglf@forum.dlang.org  
+
+Issues:  
+https://issues.dlang.org/show_bug.cgi?id=9238  
+https://issues.dlang.org/show_bug.cgi?id=8845  
+https://issues.dlang.org/show_bug.cgi?id=6221  
+https://issues.dlang.org/show_bug.cgi?id=6442  
+
+PRs:  
+https://github.com/dlang/dmd/pull/4717  
+
+## Contents
+* [Rationale](#rationale)
+* [Proposal](#proposal)
+* [Temporary destruction]()
+* [`@safe`ty implications](#safety_implications)
+* [Why not `auto ref`?](#auto_ref)
+* [Key use cases](#use_cases)
+* [Reviews](#reviews)
+
+## Rationale
+
+When calling functions that receive `ref` args, D prohibits supplying rvalues. It is suggested that this is to assist the author identifying likely logic errors where an rvalue will expire at the end of the statement, and it doesn't make much sense for a function to mutate a temporary whose life will not extend beyond the function call in question.
```

Comment:
This first sentence would work well for your opening statement in the abstract, though it could be reworded a bit, e.g. "D does not allow rvalues to be passed as function arguments where `ref` parameters are declared." or some such.